src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestAttributeValueResolver.php

```diff
@@ -27,7 +27,13 @@
      */
     public function supports(Request $request, ArgumentMetadata $argument)
     {
-        return !$argument->isVariadic() && $request->attributes->has($argument->getName());
+        if ($argument->isVariadic()) {
+            return false;
+        }
+
+        $argumentName = $argument->getName();
+
+        return $request->attributes->has($argumentName) && null !== $request->attributes->get($argumentName);
```

Comment:
After a bit of debugging, I found this in my url matcher cache:
```
        if (rtrim($pathinfo, '/') === '') {
            if (substr($pathinfo, -1) !== '/') {
                return $this->redirect($pathinfo.'/', 'app.dashboard');
            }

            return array (  'user' => NULL,  '_controller' => 'app.controller.dashboard:dashboardAction',  '_route' => 'app.dashboard',);
        }
```
It seems like the warmup of the cache adds a null value here, which is definitely not what should happen if it's not defined in the route configuration.